spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -88,15 +90,12 @@ object Engine {
    */
   def init: Unit = this.synchronized {
     if (localMode) {
-      require(getSparkMaster == null, "Detect BIGDL_LOCAL_MODE and spark.master are both set." +
-        " They're conflict. Please reset BIGDL_LOCAL_MODE if you're use Spark or remove " +
-        "spark.master if you're run without spark")
       logger.info("Detect BIGDL_LOCAL_MODE is set. Run workload without spark")
       // The physical core number should have been initialized by env variable in bigdl.sh
-      setNodeAndCore(1, getCoreNumberFromEnv)
+      setNodeAndCore(1, getNumMachineCores)
```

Comment:
@srowen I think it's a little weird if the user uses Spark master control the core number while the program will not actually use Spark...

One of the differences between spark local and the local mode is that spark local will still use spark component(e.g. RDD, BlockManager) on a single node, while local mode will directly work on arrays. See [LocalOptimizer](https://github.com/intel-analytics/BigDL/blob/master/spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/LocalOptimizer.scala). This will bring some performance difference when training some **small** models. For example, when train LeNet, the local mode can get a 4000+ throughput on my desktop, but spark local can only get 2000+. When you train some bigger model(spend more time on model forward/backward), the perf reduce become less as the overhead of spark become relatively smaller.

We still recommend the user to use spark local when development on the single node. But the local mode is still kept and used in some of our internal tests.

Ideally, I think we can use a JVM property to replace the DL_CORE_NUMBER. But keep it temperately is also fine to me.